astropy/io/fits/hdu/hdulist.py

```diff
@@ -820,6 +831,9 @@ def _readfrom(cls, fileobj=None, data=None, mode=None,
 
         saved_compression_enabled = compressed.COMPRESSION_ENABLED
 
+        # (extname, extver) tuple
+        ext = kwargs.pop("ext", None)
```

Comment:
I'd actually like a different approach altogether, but I'll update more about that in a separate comment.